### PR TITLE
Add missing interval value to unordered funnels

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

Follow up to: https://github.com/PostHog/posthog/pull/5329

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: ee/clickhouse/queries/funnels/funnel_unordered.py [hunk 1]
```diff
@@ -148,7 +148,7 @@ def _get_exclusion_condition(self):
             exclusion_time = f"exclusion_{exclusion_id}_latest_{exclusion.funnel_from_step}"
             condition = (
                 f"if( {exclusion_time} > {from_time} AND {exclusion_time} < "
-                f"if(isNull({to_time}), {from_time} + INTERVAL {self._filter.funnel_window_interval} DAY, {to_time}), 1, 0)"
+                f"if(isNull({to_time}), {from_time} + INTERVAL {self._filter.funnel_window_interval} {self._filter.funnel_window_interval_unit_ch()}, {to_time}), 1, 0)"
```
Comment (macobo) [inline ee/clickhouse/queries/funnels/funnel_unordered.py:151]: Q: Are we interpolating user input in here? :monocle_face: 

Also, wouldn't it make more sense to use https://clickhouse.tech/docs/en/sql-reference/functions/type-conversion-functions/ `toIntervalX` functions? Would interpolate nicer and allow passing things in as parameters.

Comment (neilkakkar) [reply ee/clickhouse/queries/funnels/funnel_unordered.py:151]: Not really, it's this: https://github.com/PostHog/posthog/blob/master/posthog/models/filters/mixins/funnel.py#L100

This is part of generating the syntax, and Window functions don't support arbitrary computation (even the functions you mentioned) inside the `INTERVAL` yet.

Comment (neilkakkar) [reply ee/clickhouse/queries/funnels/funnel_unordered.py:151]: What you say makes a lot of sense 👀 .

Comment (neilkakkar) [reply ee/clickhouse/queries/funnels/funnel_unordered.py:151]: We could default to the smallest interval we have (`minutes`) in the API... and pass computation conversion requirements up?.

Or keep the API the same, and do the computation ourselves to only use MINUTES in the interval?

I don't see strong arguments for either, tbh, unless you have any?

Comment (macobo) [reply ee/clickhouse/queries/funnels/funnel_unordered.py:151]: Either works for me I think - whatever helps us remove string interpolations :)

Comment (neilkakkar) [reply ee/clickhouse/queries/funnels/funnel_unordered.py:151]: Also, hmm, it's not a user string interpolation, but a restricted one. So, even keeping it as-is doesn't seem too different from the other options.

Curious, why do you want to remove this string interpolation?

Comment (macobo) [reply ee/clickhouse/queries/funnels/funnel_unordered.py:151]: Well, every sql injection story starts with interpolation. In this case it's safe now, but future refactorings might break it.

When possible parameters > injection seems like a safe play, but willing to let this go if you think this approach is better.

Comment (neilkakkar) [reply ee/clickhouse/queries/funnels/funnel_unordered.py:151]: Makes sense. Will do it as a separate PR (there's a LOT of places using the same pattern right now, and we have to deprecate the old `funnel_window_days` once the frontend is updated)

**Comment by macobo** [comment]
In other news - what problem does this solve? Is there an existing ticket? What about a test? :)

**Comment by neilkakkar** [comment]
My bad. A follow up to: https://github.com/PostHog/posthog/pull/5329 - we missed a place there. This allows us to choose arbitrary conversion windows!

**Pull request merged**